### PR TITLE
ARXIVNG-2891: fixes when trying to upload empty file

### PR DESCRIPTION
**Pull request by mhl10**
This disables the upload button when no file has been selected. In the event a user tries to bypass the UI and `POST`s an empty form, they will get redirected to the `file_upload` page with the following error message:

**Something went wrong**
No file was uploaded; please try again.

### CODE REVIEW

**Review by SBBCornell — APPROVED** [review]
Just tested and it looks good!
As a side note, the "save and continue" button still allows me to go to the next page without uploading any files but I think that counts as a separate bug.

File: submit/controllers/ui/new/upload.py [hunk 1]
```diff
@@ -577,26 +578,26 @@ def _new_file(params: MultiDict, pointer: FileStorage, session: Session,
...
+
+        loc = url_for('ui.file_upload', submission_id=submission.submission_id)
+        return {}, status.SEE_OTHER, {'Location': loc}
+
     ancillary: bool = form.ancillary.data
 
     try:
         stat = fm.add_file(upload_id, pointer, token, ancillary=ancillary)
-    except exceptions.RequestFailed as e:
+    except exceptions.RequestFailed as ex:
```
Comment (bdc34) [inline submit/controllers/ui/new/upload.py:592]: :+1:

File: submit/static/js/filewidget.js [hunk 2]
```diff
@@ -4,13 +4,17 @@
 var update_filename = function() {
     if(file.files.length > 0) {
         document.getElementById('filename').innerHTML = file.files[0].name;
+        document.getElementById('file-submit').disabled = false;
```
Comment (bdc34) [inline submit/static/js/filewidget.js:7]: :+1:

File: submit/templates/submit/file_upload.html [hunk 3]
```diff
@@ -80,7 +81,8 @@
           </label>
         </div>
         <div class="control">
-          <button type="submit" class="button is-link">Upload</button>
+          <button id="file-submit" type="submit" class="button is-link" disabled>Upload</button>
```
Comment (bdc34) [inline submit/templates/submit/file_upload.html:84]: Disabled by default when you get on the page :+1:

File: submit/controllers/ui/new/upload.py [hunk 4]
```diff
@@ -577,26 +578,26 @@ def _new_file(params: MultiDict, pointer: FileStorage, session: Session,
...
-        #                    submission_id=submission.submission_id)
-        # return {}, status.SEE_OTHER, {'Location': redirect}
-        logger.debug('Invalid form data')
-        raise BadRequest(rdata)
+        alerts.flash_failure(
+            "No file was uploaded; please try again.",
+            title="Something went wrong")
+
+        loc = url_for('ui.file_upload', submission_id=submission.submission_id)
+        return {}, status.SEE_OTHER, {'Location': loc}
```
Comment (bdc34) [inline submit/controllers/ui/new/upload.py:586]: Was the BadRequest causing problems?

The result from the controller should be handled by flow_control.py so I should be able to do a BadReqeust and still get back to the correct page with the upload form.

**Review by bdc34 — CHANGES_REQUESTED** [review]
One question about BadRequest but looks good.

File: submit/controllers/ui/new/upload.py (hunk 4 above)
Comment (mhl10) [reply submit/controllers/ui/new/upload.py:586]: Without additional logic, `BAD_REQUEST` will prevent display of any existing uploaded files.

Comment (bdc34) [reply submit/controllers/ui/new/upload.py:586]: Got it. :+1:

**Review by bdc34 — APPROVED** [review]

**Pull request merged**